cli/js/lib.deno.ns.d.ts

```diff
@@ -1365,6 +1384,7 @@ declare namespace Deno {
    *       console.log(decoder.decode(data));
    *
    * Requires `allow-read` permission. */
+  // TODO(ry) Add readFile(path, { encoding: "utf" }) returns string
```

Comment:
@ry I'll do that tomorrow

Reply:
I think `readFile(path, { decode: "utf" })` is less surprising.

Reply:
@nayeemrmn yeah I agree 👍 but... maybe we should avoid conditional return types in `Deno` namespace and just add `readFileToString/readFileStr` instead? thoughts @ry @piscisaureus

Reply:
I think the relative benefit of readFileStr is outweighed by the familiarity of the encoding: utf8 API.

Reply:
`readFileStr` is inherently familiar... it's not clear what advantage the other option has. It's just cutely shoehorning two logically different functions into one.

Reply:
@ry Said in his talk "What I regret in node": "If something is just cute, just don't do it."

Reply:
We discussed this API offline; current proposal is:
```typescript
Deno.readTextFile(
  filename: string, 
  options?: { encoding: string }
): Promise<string>;
Deno.readTextFileSync(
  filename: string, 
  options?: { encoding: string }
): string;

Deno.writeTextFile(
  filename: string, 
  content: string, 
  options?: { encoding: string }
): Promise<void>;
Deno.writeTextFileSync(
  filename: string, 
  content: string, 
  options?: { encoding: string }
): void;
```

These methods are already available in `std/fs/mod.ts`.

Reply:
Minor thing is that "text file" brings to mind a `.txt`, or something you'd give `text/plain`. Maybe `Deno.readFileText()`? I'm fine either way.